crates/subspace-farmer/src/single_disk_plot.rs

```diff
@@ -968,20 +958,27 @@ impl SingleDiskPlot {
                 }
             })?;
 
-        tasks.push(Box::pin(async move {
-            // TODO: Error handling here
-            reward_signing(rpc_client, identity).await.unwrap().await;
+        tokio::spawn({
```

Comment:
Spawning it as a separate task prevents RPC client from being destroyed deterministically, that was the reason why we persisted tasks and were waiting for it on drop.

Reply:
I can move it to `SingleDiskPlot::run` and just join on error receiving and reward signing. WDUT?